src/ledger/backend/ethereumOnChainQuerier.spec.ts

```diff
@@ -0,0 +1,155 @@
+"use strict";
+
+import { expect } from "chai";
+import { EthereumOnChainQuerier } from "./ethereumOnChainQuerier";
+import { OnChainQuerier } from "../onChainQuerier";
+
+import * as test from "#erdstall/test";
+import { Environment, setupEnv } from "test/ledger";
+import { Wallet } from "@ethersproject/wallet";
+import { IERC721Minter__factory, IERC721__factory } from ".";
+import { Tokens } from "../assets";
+import PRNG from "#erdstall/test/random";
+import { afterEach } from "mocha";
+
+const TOKEN_SIZE = 4;
+
+describe("EthereumOnChainQuerier", function () {
+	let rng: PRNG;
+	let testenv: Environment;
+	let alice: Wallet;
+	let bob: Wallet;
+	let aliceTokens: Tokens;
+	let bobTokens: Tokens;
+	let onChainQueryA: OnChainQuerier;
+	let aTransferSeed: number;
+	let bTransferSeed: number;
+	let aliceRunningTokens: string[];
+	let bobRunningTokens: string[];
+
+	before(async function () {
+		rng = test.newPrng();
+		testenv = await setupEnv(2);
+		alice = testenv.users[0];
+		bob = testenv.users[1];
+		aliceTokens = test.newRandomTokens(rng, TOKEN_SIZE);
+		bobTokens = test.newRandomTokens(rng, TOKEN_SIZE);
+
+		onChainQueryA = new EthereumOnChainQuerier(alice);
+
+		const erc721Minter = IERC721Minter__factory.connect(
+			testenv.perunArt,
+			alice,
+		);
+		for (const id of aliceTokens.value) {
+			await erc721Minter.mint(alice.address, id);
+		}
+		for (const id of bobTokens.value) {
+			await erc721Minter.mint(bob.address, id);
+		}
+	});
+
+	beforeEach(async function () {
+		aTransferSeed = rng.uInt32();
+		bTransferSeed = rng.uInt32();
+		aliceRunningTokens = [];
+		for (const aliceToken of aliceTokens.value) {
+			aliceRunningTokens.push(aliceToken.toString());
+		}
+		bobRunningTokens = [];
+		for (const bobToken of bobTokens.value) {
+			bobRunningTokens.push(bobToken.toString());
+		}
+		let ierc721 = IERC721__factory.connect(testenv.perunArt, alice);
+		for (const [index, value] of [
+			aTransferSeed % 2,
+			aTransferSeed % 3,
+			aTransferSeed % 5,
+			aTransferSeed % 7,
+		].entries()) {
```

Comment:
Maybe:
```suggestion
		for (const [index, value] of shuffleWithSeed(aTransferSeed)) {
```

With a function like:

```ts
	const shuffleWithSeed = (seed: number) => {
		return Array.from(
			{ length: TOKEN_SIZE },
			(_, i) => seed % PRIMES[i],
		).entries();
	};
```

Reply:
Instead of my proposed `shuffleWithSeed` function you can of course just map over the existing `PRIMES` array, in case you take that route.